src/track/beatstats.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <QQueue>
+
+// These classes are used to compute statistics descriptors
+// of a series of tempo values and are called from beatutils
+
+class WindowedStatistics {
+
+    QQueue<double> m_window;
+    int m_period;
+    
+    double updateWindow(double newValue) {
+        m_window.enqueue(newValue);
+        if (m_window.count() > m_period) {
+            return m_window.dequeue();
+        } else {
+            return std::nan("");
+        }
+    }
+    virtual double compute() = 0;
+    virtual void update(double newValue, double oldValue) = 0;
+
+  public:
```

Comment:
We normally put public to the beginning of the class definition and private below.